app/src/main/java/com/nervousfish/nervousfish/modules/pairing/APairingHandler.java

```diff
@@ -54,7 +56,7 @@ public void writeAllContacts() throws IOException {
      * @throws IOException When deserialization doesn't go well.
      */
     void writeContact(final Contact contact) throws IOException {
-        if (!checkExists(contact)) {
+        LOGGER.info("Begin writing contact :" + contact.getName());
             byte[] bytes = null;
```

Comment:
I fixed the indentation! The check is certainly not needed, because the methods send a contact to its client. I changed the method to sendContact for clarity. The client will have to check for himself if he accepts or rejects the contact he receives.